Expand on Neo4j and Query API (#634)
neo4j/docs-aura: modules/ROOT/pages/api/overview.adoc

The value of this header contains a unique ID that can be used to track the journey of a request.

If you run into any issues with a particular request, you can https://support.neo4j.com/[raise a support ticket] and provide the `X-Request-Id`.

> == Query API

Reviewer: Not sure if the Query API belongs in the Aura API. We could maybe change the top level header to _API_ and then have separate sections on Aura API and Query API.

The Query API allows you to interact with an Aura database using Cypher over HTTPS. 

> Unlike the deprecated HTTP API, the Query API is now exposed for Aura, making it a viable alternative to Bolt for running queries programmatically. 

Reviewer: ```suggestion
The Query API is a viable alternative to Bolt for running queries programmatically. 
```

For more information, see the link:https://neo4j.com/docs/query-api/current/[Query API documentation]

> To send Cypher queries to an Aura instance, make a POST request to the following endpoint:

Reviewer: ```suggestion
To send Cypher queries to an Aura instance, make a `POST` request to the following endpoint:
```

> https://<databaseID>.databases.neo4j.io/db/<databaseName>/query/v2

Reviewer: This needs to be escaped, it shouldn't be clickable.

`<databaseID>` The unique ID of your Aura database (e.g., e011c23c).

`<databaseName>` The target database (e.g., neo4j for the default database).

> Aura only supports HTTPS (port 443). There’s no need to specify a port explicitly, as it defaults to 443.

Reviewer: ```suggestion
Aura only supports HTTPS (port 443). 
You don't need to specify a port explicitly, as it defaults to 443.
```

> Here's how to create a node in Aura using `curl`:

Reviewer: ```suggestion
This is how to create a node in Aura using `curl`:
```

[source, shell]
----
curl -vvv -X POST
     -H "Authorization: Basic bmVvNGo6Zd9vdGJhbGg="
     -H "Content-Type: application/json"
     -d "{   \"statement\": \"create (n:Person) RETURN n;\" } }"
      https://e011c22f.databases.neo4j.io:443/db/neo4j/query/v2
----

> Note the last line above and inclusion of `:443` is not required as its the default.

Reviewer: Then why is it included?
